### ISSUE TITLE
file in/out modules do not support userDir

### ISSUE THREAD

**Issue by mrummuka** [node-red/node-red#947]
Against my expectations it seems that neither file in/out module complies with the userDir settings. Despite having userDir set 

```
30 Jul 06:30:27 - [info] Settings file  : /vagrant/data/settings.js
30 Jul 06:30:27 - [info] User directory : /vagrant/data
30 Jul 06:30:27 - [info] Flows file     : /vagrant/data/flows/flows.json
30 Jul 06:30:27 - [info] Server now running at http://127.0.0.1:1880/
```

the file seems to be read/written from/to the directory where node-red was launched from.

**Comment by knolleary** [comment]
That is true - there is no relationship between userDir and the File node.

The File nodes use the current working directory of the process by default - which is not necessarily obvious to the user depending on how they have run Node-RED.

We cannot change this behaviour as it will break existing flows. But we can look at making the existing behaviour more obvious or explicit, and perhaps adding an option for what the path should be relative to - there are scenarios where you don't know the specific path and nor should you (when running in Bluemix, for example).

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
